### DISCUSSION TITLE
Imported Namespace support like in Xtext "importedNamespace"

### DISCUSSION THREAD

**Discussion by dsmarhas** [eclipse-langium/langium#709] [Q&A]
Wondering if there the langium team is planning to support "importedNamespace" with qualified names like in XText? If not, could you please provide some pointers on how to implement this feature in my DSL?

**Comment by spoenemann** [comment]
No there is no such plan yet. Main reason: the "importedNamespace" feature of Xtext was mainly targeted at languages with namespaces (qualified names), like Java packages. That feature offered automatic conversion of qualified names so they could be accessed with simple name after importing. This is a very special case that we don't want to include in the core functionality of Langium.

Note that all exported symbols are globally accessible by default. If you want to change that, you can customize the `ScopeProvider` service. Alternatively, you can keep them accessible in the scope, but add a validation rule that forbids using a symbol that is not explicitly imported; advantage of this approach: more descriptive error message, easier to provide a quick fix (code action), navigation etc. works even for non-imported symbols.
